Userland/Libraries/LibJS/Runtime/Completion.h

```diff
@@ -94,11 +100,140 @@ class [[nodiscard]] Completion {
     }
 
 private:
+    class EmptyTag {
+    };
+    friend AK::Optional<Completion>;
+
+    Completion(EmptyTag)
+        : m_type(Type::Empty)
+    {
+    }
+
+    bool is_empty() const
+    {
+        return m_type == Type::Empty;
+    }
+
     Type m_type { Type::Normal }; // [[Type]]
     Optional<Value> m_value;      // [[Value]]
     Optional<FlyString> m_target; // [[Target]]
 };
 
+}
+
+namespace AK {
+
+template<>
+class Optional<JS::Completion> {
+    template<typename U>
+    friend class Optional;
+
+public:
```

Comment:
This optimization is somewhat optional I like it a bit less since Completion is holding an `Optional<FlyString>` anyway.
However with the max recursion test it does increase it from `1661` to `1716`.